modules/core/src/main/java/org/locationtech/jts/geom/Coordinates.java

```diff
@@ -46,7 +48,100 @@ public static Coordinate create(int dimension, int measures)
     }
     return new Coordinate();
   }
-    
+
+  /**
+   * Utility method ensuring array contents are of the specified dimension and measures.
+   * <p>
+   * Array is modified in place if required, coordinates are replaced in the array as required
+   * to ensure all coordinates have the same dimension and measures.
+   * </p>
+   *
+   * @param array
+   * @return array returned, or copy created if required to enforce consistency.
+   */
+  public static Coordinate[] enforceConsistency(Coordinate[] array)
```

Comment:
test case?  Or is it covered elsewhere?

Reply:
I will add a test case, thanks! Previously it was covered by other tests